.svglintrc.js

```diff
@@ -80,6 +81,33 @@ module.exports = {
               reporter.error("Unexpected character(s) detected outside the opening and/or closing <svg> tags");
             }
           },
+          function(reporter, $, ast) {
+            reporter.name = "icon-centered";
+            const iconPath = $.find("path").attr("d");
+            const bounds = getBounds(iconPath);
+
+            if (
+              iconIgnored.size.hasOwnProperty(iconPath) ||
+              iconIgnored.center.hasOwnProperty(iconPath)
+            ) {
+              return
+            }
+
+            const targetCenter = iconSize / 2;
+            const centerX = +((bounds.minX + bounds.maxX) / 2).toFixed(iconFloatPrecision);
+            const devianceX = centerX - targetCenter;
+            const centerY = +((bounds.minY + bounds.maxY) / 2).toFixed(iconFloatPrecision);
+            const devianceY = centerY - targetCenter;
+
+            if (
+              devianceX < -iconTolerance ||
+              devianceX > iconTolerance ||
+              devianceY < -iconTolerance ||
+              devianceY > iconTolerance
+            ) {
+              reporter.error(`<path> must be centered on the canvas; the center is currently ${centerX} x ${centerY}`);
```

Comment:
No real opinion on that, but maybe write how much it differs (`devianceX/Y`) instead of the current center?
But again, I'm not even sure what I would prefer.

Reply:
I'm not against this, but I'm not sure how useful that would be in practice. I'd say if we find it would be useful we can always add it later?

Reply:
Yeah, sure! :)
Like I said, it's not really a must. I think both values say basically the same, but with the deviance you can see on one glance where something has gone wrong, and in which direction. The center value just needs a tiiiiiny bit of more brain time to process, I guess.

Reply:
On a slight tangent, it could be hard to ascertain that the centre should be 12,12 (e.g. one either needs to know the size should be 24 x 24 and figure it out or already know it should be 12,12 -- details of which are in the contributor guide but still...). When I was writing the PR, it's reasonably clear to see from the lint output of 203 erroring icons what the value should probably be:

```sh
[...]
x icons/zalando.svg
  x icon-centered <path> must be centered on the canvas; the center is currently 11.509 x 12
x icons/zdf.svg
  x icon-centered <path> must be centered on the canvas; the center is currently 12 x 11.998
x icons/zoom.svg
  x icon-centered <path> must be centered on the canvas; the center is currently 12 x 12.024
```

but you lose this point of reference as these icons are being ignored and eventually, when icons are fixed, a contributor _should_ only be seeing one error at a time.  If the centre is a long way out, there's some effort needed to parse something like:

```sh
x icons/fakeicon.svg
  x icon-centered <path> must be centered on the canvas; the center is currently 11.444 x 12.638
```

Anyway, I agree with @runxel that knowing where and in which dimension something's gone wrong is useful. At the same time I feel like seeing the expected value + problematic value together explains it in absolute terms so
I've pushed a change to the output to indicate what the centre should be, plus a formatting change to use coordinate format versus `w x h` dimensions.  I don't know how this output goes for countries that use the comma as a decimal separator; I added a space to hopefully help in that case + for readability.    See what you think 👍   

```sh
x icons/zoom.svg
  x icon-centered <path> must be centered at (12, 12); the center is currently (12, 12.024)
```

Reply:
I'm completely fine with the current error message @davidjb, nice work.

Regarding "the comma as a decimal separator": the space does make that clear but as far as I know programming languages (and by extension programmers) always use a dot as a decimal separator so it shouldn't be a problem to begin with.

Reply:
Cool, thanks @ericcornelissen.  I was aiming for as explicit as possible, and as a plus it seems more readable/scannable with spaces.